pb/protocol_messenger.go

```diff
@@ -0,0 +1,174 @@
+package dht_pb
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/libp2p/go-libp2p-core/host"
+	"github.com/libp2p/go-libp2p-core/peer"
+	"github.com/libp2p/go-libp2p-core/routing"
+
+	logging "github.com/ipfs/go-log"
+	record "github.com/libp2p/go-libp2p-record"
+	recpb "github.com/libp2p/go-libp2p-record/pb"
+	"github.com/multiformats/go-multihash"
+
+	"github.com/libp2p/go-libp2p-kad-dht/internal"
+)
+
+var logger = logging.Logger("dht")
+
+// ProtocolMessenger can be used for sending DHT messages to peers and processing their responses.
+// This decouples the wire protocol format from both the DHT protocol implementation and from the implementation of the
+// routing.Routing interface.
+//
+// Note: the ProtocolMessenger's MessageSender still needs to deal with some wire protocol details such as using
```

Comment:
that this `MessageSender` is different from the `messageSender` used by the `message manager` is somewhat confusing. any possibility of having different names for the two?

Reply:
Good catch.

I renamed `messageSender -> peerMessageSender` and clarified that it sends messages to a particular peer. We could rename `messageManager` to `messageSender` as the default implementation of the `MessageSender` interface, or just leave it named as is (or some better name) WDYT?

Reply:
seems good to me 👍

Reply:
Done and as you suggested offline renamed `messageManager` to `messageSenderImpl`